ci/etc/valgrind-cran.supp

```diff
@@ -32,3 +32,21 @@
    fun:getvar
    fun:bcEval
 }
+
+{
+   # This also doesn't seem to cause issues on CRAN, so suppress it.
+   Memcheck:Leak
+   match-leak-kinds: possible
+   fun:malloc
+   fun:libdeflate_alloc_compressor
+   fun:do_memCompress
+   fun:bcEval_loop
+   fun:bcEval
+   fun:Rf_eval
+   fun:R_execClosure
+   fun:applyClosure_core
+   fun:Rf_applyClosure
+   fun:Rf_eval
+   fun:do_set
+   fun:Rf_eval
+}
```

Comment:
We've had valgrind warnings for a while now (since before we submitted last IIRC), and if so, they aren't bugging the cran version of valgrind, so I've added to the suppression file. We might need to be careful and come back and remove this if the next version of arrow gets dinged fro valgrind errors. But this hopefully will get the job to pass.